### PR TITLE
wgsl: Add `scalarRange` and `sparseScalarRange` to FPTraits API

### PR DESCRIPTION
**Pull request by zoddicus**
There is also a bunch of refactoring of fullF*Range to scalarF*Range
and such to make things clearer, since it always confuses people that
fullF32Range only returns a subset of f32.

Issue #3174

<hr>

**Requirements for PR author:**

- [x] All missing test coverage is tracked with "TODO" or `.unimplemented()`.
- [x] New helpers are `/** documented */` and new helper files are found in `helper_index.txt`.
- [x] Test behaves as expected in a WebGPU implementation. (If not passing, explain above.)

**Requirements for [reviewer sign-off](https://github.com/gpuweb/cts/blob/main/docs/reviews.md):**

- [ ] Tests are properly located in the test tree.
- [ ] [Test descriptions](https://github.com/gpuweb/cts/blob/main/docs/intro/plans.md) allow a reader to "read only the test plans and evaluate coverage completeness", and accurately reflect the test code.
- [ ] Tests provide complete coverage (including validation control cases). **Missing coverage MUST be covered by TODOs.**
- [ ] Helpers and types promote readability and maintainability.

When landing this PR, be sure to make any necessary issue status updates.

### CODE REVIEW

**Comment by zoddicus** [comment]
This is written atop of my vector range change, so all the new code is in 191d605923d0b0fd13527713ba39d9a9a01bf047

File: src/webgpu/shader/execution/expression/call/builtin/degrees.spec.ts [hunk 1]
```diff
@@ -11,38 +11,32 @@ import { makeTestGroup } from '../../../../../../common/framework/test_group.js'
...
 import { FP } from '../../../../../util/floating_point.js';
-import { fullF16Range, fullF32Range, fullF64Range } from '../../../../../util/math.js';
 import { makeCaseCache } from '../../case_cache.js';
 import { allInputSources, onlyConstInputSource, run } from '../../expression.js';
 
 import { abstractBuiltin, builtin } from './builtin.js';
 
 export const g = makeTestGroup(GPUTest);
 
+const cases = (['f32', 'f16'] as const)
```
Comment (jzm-intel) [inline src/webgpu/shader/execution/expression/call/builtin/degrees.spec.ts:21]: Maybe
``` suggestion
const f32_f16_cases = (['f32', 'f16'] as const)
```

File: src/webgpu/shader/execution/expression/call/builtin/mix.spec.ts [hunk 2]
```diff
@@ -34,8 +49,7 @@ const vec_scalar_cases = (['f32', 'f16'] as const)
           return FP[trait].generateVectorPairScalarToVectorComponentWiseCase(
             FP[trait].sparseVectorRange(dim),
             FP[trait].sparseVectorRange(dim),
-            // NB: Refactor this when adding scalar ranges to FPTrait API
-            trait === 'f32' ? sparseF32Range() : sparseF16Range(),
+            FP[trait].sparseScalarRange(),
             nonConst ? 'unfiltered' : 'finite',
             ...FP.f32.mixIntervals
```
Comment (jzm-intel) [inline src/webgpu/shader/execution/expression/call/builtin/mix.spec.ts:54]: This line should be `FP[trait].mixIntervals` and I have comment on the first PR.

**Review by jzm-intel — APPROVED** [review]
Looks good to me, thanks!

File: src/webgpu/shader/execution/expression/call/builtin/degrees.spec.ts (hunk 1 above)
Comment (zoddicus) [reply src/webgpu/shader/execution/expression/call/builtin/degrees.spec.ts]: Merged the abstract generator in, generating an empty array of Cases when non_const.

File: src/webgpu/shader/execution/expression/call/builtin/mix.spec.ts (hunk 2 above)
Comment (zoddicus) [reply src/webgpu/shader/execution/expression/call/builtin/mix.spec.ts]: Done

File: src/webgpu/util/math.ts [hunk 3]
```diff
@@ -1605,8 +1605,8 @@ const kSparseVectorF16Values = {
  *
  * This is an even more stripped down version of `vectorF16Range` for when
  * pairs of vectors are being tested.
- * All of the interesting floats from sparseF16 are guaranteed to be tested, but
- * not in every position.
+ * All of the interesting floats from sparseScalarF16 are guaranteed to be
+ * tested, butnot in every position.
```
[the remark this answers is not in the document]
Comment (zoddicus) [reply src/webgpu/util/math.ts]: Done

**Pull request merged**